sycl/plugins/cuda/CMakeLists.txt

```diff
@@ -21,6 +21,10 @@ add_library(pi_cuda SHARED
   "pi_cuda.cpp"
 )
 
+add_dependencies(pi_cuda
+  ocl-headers
+)
+
```

Comment:
Could you remove `OpenCL-Headers` from the line 36, please?
```
target_link_libraries(pi_cuda PUBLIC OpenCL-Headers cudadrv)
```

Reply:
I tried this
```diff
diff --git a/sycl/plugins/cuda/CMakeLists.txt b/sycl/plugins/cuda/CMakeLists.txt
index 0b8b5e30233..01aced8d380 100644
--- a/sycl/plugins/cuda/CMakeLists.txt
+++ b/sycl/plugins/cuda/CMakeLists.txt
@@ -33,7 +33,7 @@ target_include_directories(pi_cuda PRIVATE "${sycl_inc_dir}")
 
 target_include_directories(pi_cuda INTERFACE ${CUDA_INCLUDE_DIRS})
 
-target_link_libraries(pi_cuda PUBLIC OpenCL-Headers cudadrv)
+target_link_libraries(pi_cuda PUBLIC cudadrv)
 
 target_link_libraries(sycl INTERFACE pi_cuda)
 

```

and the build broke again in the same way as before.

Reply:
Did you tried this w/o adding
```cmake
add_dependencies(pi_cuda
  ocl-headers
)
```
?

Reply:
Comparing the build directories, the difference is that `pi_cuda` no longer adds `tools/sycl/OpenCL/inc` to the include path.
So, even if it is built after the OpenCL headers are available (because the dependency is there), it doesn't pick them up.